### ISSUE TITLE
separate the seedpassword from the seed

### ISSUE THREAD

**Issue by robvanmieghem** [rivine/rivine#119]
currently a wallet is initialized an the seed is encrypted using the seed itself as a password, It should be standard to use a different encryptionkey than the data you are encrypting

**Comment by GlenDC** [comment]
@robvanmieghem it is already supported to use a password by using the `-p/--password` flag, as you already know. So do you mean that you want to make a password required? If so, do you want to promote it to a positional argument. Can you please elaborate? As your issue description sounds as if a custom password isn't supported, while it in fact is.

**Comment by GlenDC** [comment]
@robvanmieghem I think we should require a password. We can rely on the `speakeasy` lib to ask for a passphrase/password, as creating a password is a manual process anyhow. However, I think it cannot hurt if we also keep it as a flag, such that if the flag is given we simply use it without having to ask for it.

Not sure if we should enforce a password to be non-empty. Other blockchain clients (e.g. `geth`) allow this, and I don't see why not. In the end there is no big difference between an empty pass and a 0 pass. And it's not like it's a default, as the user would have to choose for it to be empty (remember if `-p` is not given, a password is asked using the `speakeasy` lib, so only if even then the user still wants to go for an empty pass, would it be empty.

In short, I think we should make it obvious to the user that she should choose a pass, by asking for it explicitly using `speakeasy`.

**Comment by lafont-e** [comment]
As the request for 119 seems very simple, I need to explain what I understand
```
CURRENT SITUACION
a) if no -p flag is passed ==> seed is used as password 
b) if -p flag is used ==> ask for password ==> use asked password as password 

WANTED SITUATION
a) if -p flag password is used ==> ask for password ==> use asked password as password (no change)
b) if no -p flag is passed ==> ask for password ==> use asked password as password
c) Asked passwords may be empty only if the user did not select the -p flag
```

So, I changed WalletInitcmd to:
```go
// Walletinitcmd encrypts the wallet with the given password
func Walletinitcmd() {
	var er api.WalletInitPOST

	qs := fmt.Sprintf("dictionary=%s", "english")

	if !initPassword {
		fmt.Println("You should provide a password, it may be empty if you wish.")
	}

	for {
		password, err := speakeasy.Ask("Wallet password: ")
		if err != nil {
			Die("Reading password failed:", err)
		}

		if !initPassword || password != "" {
			break
		}

		fmt.Println("The password cannot be empty. Please enter a password.")
	}

	if password != "" {
		qs += fmt.Sprintf("&encryptionpassword=%s", password)
	}
	
	err := PostResp("/wallet/init", qs, &er)
	if err != nil {
		Die("Error when encrypting wallet:", err)
	}

	fmt.Printf("Recovery seed:\n%s\n\n", er.PrimarySeed)
	fmt.Printf("Wallet encrypted with given password\n")
}
```
If you agree, I will push it.
